core/controllers/email_dashboard.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2016 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controller for user query related pages and handlers."""
+
+from core.controllers import base
+from core.domain import config_domain
+from core.domain import user_query_services
+from core.domain import user_query_jobs_one_off
+from core.domain import user_services
+from core.platform import models
+
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+
+current_user_services = models.Registry.import_current_user_services()
+cursor_services = models.Registry.import_cursor_services()
```

Comment:
Do we really need this new abstraction, though? Would it be possible, instead, to pass the urlsafe cursor into a new method in the gae_models class which acts as a seam to fetch_page() -- and pass a urlsafe cursor out from that method as well? That would still preserve the separation, but reduce the amount of code "surface" that needs to be maintained.